src/test/kotlin/com/exactpro/th2/sim/SimRuleInfoTest.kt

```diff
@@ -0,0 +1,221 @@
+package com.exactpro.th2.sim
+
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.event.EventUtils.createMessageBean
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventStatus
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.grpc.MessageGroup
+import com.exactpro.th2.common.grpc.MessageGroupBatch
+import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.message.addField
+import com.exactpro.th2.common.message.getString
+import com.exactpro.th2.common.message.message
+import com.exactpro.th2.common.message.messageType
+import com.exactpro.th2.common.message.plusAssign
+import com.exactpro.th2.common.message.sessionAlias
+import com.exactpro.th2.common.schema.message.MessageRouter
+import com.exactpro.th2.sim.configuration.RuleConfiguration
+import com.exactpro.th2.sim.impl.SimulatorRuleInfo
+import com.exactpro.th2.sim.rule.IRule
+import com.exactpro.th2.sim.rule.IRuleContext
+import com.google.protobuf.ByteString
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.mockito.kotlin.check
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.reset
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.time.Instant
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledThreadPoolExecutor
+
+class SimRuleInfoTest {
+
+    @Test
+    fun `send parsed`() {
+        val batchRouter = mock<MessageRouter<MessageGroupBatch>>()
+        val eventRouter = mock<MessageRouter<EventBatch>>()
+
+        val testParsedMessage = message("SomeMessage").apply {
+            metadataBuilder.protocol = "TestProtocol"
+            addField("SomeField", "SomeValue")
+        }.build()
+
+        fun MessageGroupBatch.check() {
+            val message = this.getGroups(0).getMessages(0).message
+            Assertions.assertEquals("SomeMessage", message.messageType)
+            Assertions.assertEquals(testAlias, message.sessionAlias)
+            Assertions.assertEquals("TestProtocol", message.metadata.protocol)
+            Assertions.assertEquals("SomeValue", message.getString("SomeField"))
+        }
+
+        SimulatorRuleInfo(0, EmptyTestRule, ruleConfiguration, batchRouter, eventRouter, rootEventId, scheduler) {
+            LOGGER.debug("Rule removed")
+        }.let { simulatorRuleInfo ->
+
+            simulatorRuleInfo.send(testParsedMessage)
+            verify(batchRouter, times(1)).sendAll(check(MessageGroupBatch::check), check { Assertions.assertEquals("second", it) }, check { Assertions.assertEquals(testRelation, it) })
```

Comment:
Why don't use `eq` method from mockito?
```suggestion
            verify(batchRouter, times(1)).sendAll(check(MessageGroupBatch::check), eq("second"), eq(testRelation))
```